consensus/src/chained_bft/event_processor.rs

```diff
@@ -609,10 +609,20 @@ impl<T: Payload> EventProcessor<T> {
 
     /// Generate sync info that can be attached to an outgoing message
     fn gen_sync_info(&self) -> SyncInfo {
+        let hqc = self.block_store.highest_quorum_cert().as_ref().clone();
+        // No need to include HTC if it's lower than HQC
+        let hqc_round = hqc.certified_block_round();
+        let htc = self.pacemaker.highest_timeout_certificate().and_then(|tc| {
+            if tc.round() > hqc_round {
+                Some(tc)
+            } else {
+                None
+            }
```

Comment:
you can just
```rust 
        let htc = self.pacemaker.highest_timeout_certificate().filter(|tc| tc.round > hqc.round);
```

Reply:
Lol I knew there should be a simpler way to do it, thanks!